src/util/util_drs.cpp

```diff
@@ -0,0 +1,64 @@
+#include "util_drs.h"
+#include "util_env.h"
+
+namespace dxvk::drs {
+    bool parsedrssetting(const std::string_view& str, NvU32& setting) {
+        if (auto it = settings.find(str); it != settings.end()) {
+            setting = it->second;
+            return true;
+        } else {
+            return str::parsedword(str, setting);
+        }
+    }
+
+    bool parsedrsdwordvalue(NvU32 key, const std::string_view& str, NvU32& value) {
+        if (auto it = values.find(key); it != values.end()) {
+            auto& values = it->second;
+            if (auto it = values.find(str); it != values.end()) {
+                value = it->second;
+                return true;
+            }
+        }
+
+        return str::parsedword(str, value);
+    }
+
+    std::unordered_map<NvU32, NvU32> parsedrsdwordsettings(const std::string& str) {
+        std::unordered_map<NvU32, NvU32> result;
+
+        if (str.empty())
+            return result;
+
+        auto entries = str::split<std::vector<std::string_view>>(str, std::regex(","));
+
+        for (auto entry : entries) {
+            auto eq = entry.find('=');
+
+            if (eq == entry.npos || eq == 0 || eq == entry.size() - 1)
+                continue;
+
+            NvU32 key, value;
+
+            if (parsedrssetting(entry.substr(0, eq), key) && parsedrsdwordvalue(key, entry.substr(eq + 1), value))
```

Comment:
Since this is now specific to `drs`, we could log/trace parsing errors for key/value here. That said, we already show the summary, so this would be quite verbose. So no real preference on my side.

Reply:
I have a slight preference towards avoiding log statements in "util" code, and yeah, logging what was parsed successfully is probably good enough.